src/gsi/core_disp_func.F90

```diff
@@ -0,0 +1,1055 @@
+#ifdef FCA_REF_MOD
+module core_disp_func_m
+!$$$ module documentation block
+!           .      .    .                                       .
+! module:   core_disp_func_m
+!  prgmmr: Nehrkorn
+!
+! abstract: Core displacement functionality routines, nonlinear (NLM) version
+!
+! program history log:
+!
+! subroutines included:
+!   sub compute_xy_orig
+!   sub compute_xy_orig_dm
+!   sub exchange_needed
+!   sub exchange_interpolated
+!   sub apply_disp_2d
+!   SUB APPLY_VERT
+!   sub allocate_disp
+!   sub deallocate_disp
+!
+! functions included:
+!   bicub_interp
+!   cubicInterpolate
+!   bilin_interp
+!   init_meta
+!
+! variable definition:
+!
+! attributes:
+!   language: f90
+!   machine:
+!
+!$$$ end documentation block
+  use fp_types_m, only: fp
```

Comment:
Please don't add more floating point types.  All GSI routines should use integer lengths and real precisions from the kinds module.

use kinds, only: r_kind, i_kind

Reply:
Done.